android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -349,10 +349,10 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             @Override
             public void afterTextChanged(Editable s) {
                 doneButton.setImageResource(R.drawable.ic_done_grey);
-                if (widgetArgs.getJsonObject().optBoolean(JsonFormConstants.EXPAND_ON_TEXT_CHANGE))
-                    addOnDoneAction(referenceEditText, doneButton, widgetArgs);
                 ValidationStatus validationStatus = JsonFormFragmentPresenter
                         .validate(widgetArgs.getFormFragment(), referenceEditText, false);
+                if (widgetArgs.getJsonObject().optBoolean(JsonFormConstants.EXPAND_ON_TEXT_CHANGE) && validationStatus.isValid())
```

Comment:
I think it's better to simple move https://github.com/OpenSRP/opensrp-client-native-form/pull/509/commits/5a3d1f74eaee2f3288018b20a00145769a07d218#diff-5faa6441c13d21a9d335d1851497f620c839a7c50ec84a6b1b77b96892e94d1aL352 into this if block https://github.com/OpenSRP/opensrp-client-native-form/pull/509/commits/5a3d1f74eaee2f3288018b20a00145769a07d218#diff-5faa6441c13d21a9d335d1851497f620c839a7c50ec84a6b1b77b96892e94d1aR355.